vyper/parser/expr.py

```diff
@@ -419,23 +524,43 @@ def parse_BinOp(self):
                 return
             new_typ = BaseType(ltyp)
 
-            if ltyp == rtyp == "uint256":
-                arith = [
-                    "seq",
-                    [
-                        "assert",
-                        [
-                            "or",
-                            # r == 1 | iszero(r)
-                            # could be simplified to ~(r & 1)
-                            ["or", ["eq", "r", 1], ["iszero", "r"]],
-                            ["lt", "l", ["exp", "l", "r"]],
-                        ],
-                    ],
-                    ["exp", "l", "r"],
-                ]
-            elif ltyp == rtyp == "int128":
-                arith = ["exp", "l", "r"]
+            if self.expr.left.get("value") == 1:
+                return LLLnode.from_list([1], typ=new_typ, pos=pos)
+            if self.expr.left.get("value") == 0:
+                return LLLnode.from_list(["iszero", right], typ=new_typ, pos=pos)
+
+            if ltyp == "int128":
+                is_signed = True
+                num_bits = 128
+            else:
+                is_signed = False
+                num_bits = 256
+
+            if isinstance(self.expr.left, vy_ast.Int):
+                value = self.expr.left.value
+                upper_bound = calculate_largest_power(value, num_bits, is_signed) + 1
+                if is_signed:
+                    clamp = ["and", ["slt", right, upper_bound], ["sgt", right, -upper_bound]]
```

Comment:
Can't use a negative power
```suggestion
                    clamp = ["and", ["slt", right, upper_bound], ["sgt", right, -1]]
```

Reply:
Good catch!